python/__init__.py.in

```diff
@@ -95,3 +95,20 @@ def add_deprecation_warnings():
 
 
 add_deprecation_warnings()
+
+
+################################################################################
+# add support for IPython
+
+
+try:
+    from IPython import get_ipython
+
+    def f3d_image_repr_png(f3d_img: Image):
+        return f3d_img.save_buffer(format=Image.SaveFormat.PNG)
+
+    if ipython := get_ipython():
+        png_formatter = ipython.display_formatter.formatters["image/png"]
+        png_formatter.for_type(Image, f3d_image_repr_png)
+except ImportError:
+    pass
```

Comment:
we do this do ? is this tested ?

Reply:
It attempts to import `IPython` and register a handler to display `f3d.Image` instances as PNG images, if the import fail then we're not in an notebook so we don't care and do nothing (python uses `pass` as a no-op because you can't otherwise have empty blocks as it's indent-based).

Actually there's an alternative implementation to consider, I'll add a comment

Reply:
Only tested locally in a Jupyter notebook. I'll check if I can add something using pytest but I guess we'll have to add ipython in our local python install.  
@snoyer any suggestion about a simple unit test using IPython is welcomed

Reply:
Short of rendering a notebook to html/markdown/pdf and checking that the output file actually contains an image I'm not sure we can really test it, and at that point it's no really unit testing anymore.

Going with the alternative `_repr_png_` implementation I think it'd be enough to trust that ipython/jupyter does call `_repr_png_` as advertised and only unit test `assert img._repr_png_() == f3d_img.save_buffer(format=Image.SaveFormat.PNG)`